R/teams.R

```diff
@@ -95,3 +95,68 @@ setMethod("delete", "CrunchTeam", function(x, ...) {
     dropCache(absoluteURL("../", u))
     invisible(out)
 })
+
+#' Share Crunch assets with a team
+#' 
+#' You can share filters and multitables with a team that you are on. This will 
+#' give all team members access to view and edit these filters. Use `getTeams()`
+#' to see what teams you are on.
+#' 
+#' @param x a `CrunchFilter` or `Multitable`
+#' @param value a `CrunchTeam` or url for a Crunch team
+#' 
+#' @return a `CrunchTeam` that the asset is shared with.
+#' 
+#' @rdname team-sharing
+#' @export
+setGeneric("team", function(x) standardGeneric("team"))
```

Comment:
These might not need to be methods, we could instead do something like
```
team <- function(x) {
    stopifnot(class(x) %in% c("CrunchFilter", "Multitable"))
   ....
}
```

This would both remove some code and also make it easier to add new assets to the team structure once they get added.

Reply:
I'm inclined to leave them as functions since we want the dispatch based on the type of `value` (I know we could add an if/else inside the function, but I think this is a bit more elegant). I have collapsed them into single helper functions which saves a few characters.